PlayList.java

```diff
@@ -35,32 +35,59 @@ public Track getTrack(int index) {
      *  If the list is full, does nothing and returns false.
      *  Otherwise, appends the track and returns true. */
     public boolean add(Track track) {
-        //// replace the following statement with your code
+        int size1 = this.getSize();
+        if(size1 == this.getMaxSize()){
```

Comment:
Wouldn't it be better to use `this.size` directly?
```suggestion
        if(this.size == this.maxSize){
```